crates/ironrdp-server/src/encoder/mod.rs

```diff
@@ -0,0 +1,86 @@
+pub mod bitmap;
+
+use ironrdp_pdu::{
+    cursor::WriteCursor,
+    fast_path::{EncryptionFlags, FastPathHeader, FastPathUpdatePdu, Fragmentation, UpdateCode},
+    PduEncode,
+};
+
+use self::bitmap::BitmapEncoder;
+
+use super::BitmapUpdate;
+
+const MAX_FASTPATH_UPDATE_SIZE: usize = 16_374;
+
+pub struct UpdateEncoder {
+    buffer: Vec<u8>,
+    bitmap: BitmapEncoder,
+}
+
+impl UpdateEncoder {
+    pub fn new() -> Self {
+        Self {
+            buffer: vec![0; 8192 * 8192],
```

Comment:
Same comment as in `crates/ironrdp-server/src/encoder/bitmap.rs`, we need to think of a way to make encode buffer size not hard coded.

Reply:
For this buffer, a solution I can see is starting with a small size and doing a manual doubling of the buffer when it is not enough.
Something like this:

```rust
let len = loop {
    match self.bitmap.encode(&bitmap, self.buffer.as_mut_slice()) {
        Err(e) => match e.kind() {
            ironrdp_pdu::PduErrorKind::NotEnoughBytes { .. } => {
                self.buffer.resize(self.buffer.len() * 2, 0);
            }

            _ => return None;
        },
        Ok(len) => break len,
    }
};
```

What do you think?

Reply:
It’s probably good enough for now. I’m thinking in the future we could use the [`WriteBuf`](https://github.com/Devolutions/IronRDP/blob/2eedfe7b8d987d25ec3996738c69d1d0bb3d416d/crates/ironrdp-pdu/src/write_buf.rs) (it’s not yet merged on master)